src/north/CsvToHttp/CsvToHttp.class.js

```diff
@@ -0,0 +1,220 @@
+const Papa = require('papaparse')
+const fs = require('fs')
+
+const ApiHandler = require('../ApiHandler.class')
+
+/**
+ * Class CsvToHttp - convert a csv file into http request such as POST/PUT/PACTH
+ */
+class CsvToHttp extends ApiHandler {
+  /**
+   * Constructor for CsvToHttp
+   * @constructor
+   * @param {Object} applicationParameters - The application parameters
+   * @param {Engine} engine - The Engine
+   * @return {void}
+   */
+  constructor(applicationParameters, engine) {
+    super(applicationParameters, engine)
+
+    const { applicativeHostUrl, requestMethod, bodyMaxLength, authentication, csvDelimiter, mapping, proxy = null } = applicationParameters.CsvToHttp
+
+    // Creation of an entity request with is composed of all necessaries informations
+    this.request = {
+      host: applicativeHostUrl,
+      method: requestMethod,
+      authenticationField: authentication,
+    }
+
+    this.bodyMaxLength = bodyMaxLength
+
+    this.mapping = mapping || {}
+    this.csvDelimiter = csvDelimiter
+    this.proxy = this.getProxy(proxy)
+
+    this.httpBody = []
```

Comment:
httpBody should not be a class variable : you use it like a local variable in handleFile which should be the returned value of the convertToHttpBody function and then should be used as an argument for your sendData function